README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# monash

<!-- badges: start -->
<!-- badges: end -->

The `monash` R-package is a utility package with consolidated tools and
templates for staffs at Monash University. The package contains three
primary themes:

-   **workflow**: automating repetitive tasks, particularly teaching
    related activities;
-   **template**: Monash branded R Markdown templates for teaching,
    presentation, etc; and
-   **data**: easy access to relevant information (may require access
    privileges).

## Installation

You can install `monash` R-package as below:

``` r
# install.packages("remotes")
remotes::install_github("numbats/monash")
```

## Get Monash logo

You can get a copy of the logo into the directory you wish with below.

``` r
# default logo
monash::logo_get(path = "man/figures")
#> ✔ The 'monash-stacked-blue-rgb.png' is now in 'man/figures/'
#> man/figures/monash-stacked-blue-rgb.png
# black and white & one-line version of the logo
monash::logo_get(path = "man/figures", color = "black", stack = FALSE, filename = "monash-logo-black.png")
#> ✔ The 'monash-logo-black.png' is now in 'man/figures/'
#> man/figures/monash-logo-black.png
```

And then you can reference the logo file that you copied.

![](man/figures/monash-stacked-blue-rgb.png)
![](man/figures/monash-logo-black.png)

## Get Monash brand colors

These are handy commands to quickly see Monash brand colors and be able
to copy-and-paste the hex color codes.

``` r
monash::color_show()
```

<img src="man/figures/README-unnamed-chunk-2-1.png" width="100%" />

    #>      blue     black     white    gray80    gray50    gray10      blue    purple 
    #> "#006DAE" "#000000" "#FFFFFF" "#5A5A5A" "#969696" "#E6E6E6" "#027EB6" "#746FB2" 
    #>   fuchsia      ruby      pink       red    orange     umber     olive     green 
    #> "#9651A0" "#C8008F" "#ee64a4" "#EE0220" "#D93F00" "#795549" "#6F7C4D" "#008A25"

## Settings

(WIP) The monash package makes use of some values, listed below, from
your R profile. You can modify this by using `usethis::edit_r_profile()`
and adding below with values modified to your own values.

    options(monash.full_name = "Emi Tanaka",
            monash.email = "emi.tanaka@monash.edu",
            monash.orgunit = "Department of Econometrics and Business Statistics",
            monash.teaching_dir = "~/teaching/monash/",
            monash.workshop_dir = "~/workshop/")

## Teaching

(WIP) `create_teaching_directory("ETCXXXX")` would create a skeleton
directory with folder and file structure below for development of
teaching material. What is the advantage of this structure? It’s
designed to be synced with a private github repository and the materials
in the `release` folder are *automatically* pushed using Github
Repository to the public github repository for public release. The
website associated with it will be also be built automatically as well.
The website format will closely resemble [ETC5512: Wild-Caught
Data](https://wcd.numbat.space/).

(WIP) `release_tutorial(1)` moves `tutorial-01.html` (and .Rmd) to the
appropriate place in the `release` folder.

(WIP) Tutorial, slides and other templates for teaching.

    ├── ETCXXXX.Rproj
    ├── .git
    ├── data
    │   └── data01.csv
    ├── lectures
    │   ├── assets
    │   │   ├── monash-brand.css
    │   │   └── monash-font.css  
    │   ├── images
    │   │   ├── lecture-01-image01.png
    │   │   └── lecture-02-image01.png  
    │   ├── lecture-01.Rmd
    │   ├── lecture-02.Rmd
    │   └── lecture-03.Rmd
    ├── tutorials
    │   ├── images
    │   │   ├── tutorial-01-image01.png
    │   │   └── tutorial-02-image01.png 
    │   ├── tutorial-01.Rmd
    ├── assessments
    │   ├── assignment-01.Rmd
    │   └── quiz-01.Rmd
    ├── release # for public release
    │   ├── site
    │   │   └── _site.yml
    │   ├── data
    │   ├── lectures
    │   ├── tutorials


NEWS.md
# monash 0.0.0.9000

* Added `upload` argument to `zoom_attendance()` for automatically uploading
  to the spreadsheet.
* Added a `NEWS.md` file to track changes to the package.


R/brand.R
#' Monash brand colors
#'
#' @name monash-color
#' @export
color_primary_show <- function() {
  scales::show_col(color_primary())
  color_primary()
}

#' @family
#' @export
color_secondary_show <- function() {
  scales::show_col(color_secondary())
  color_secondary()
}


#' @export
color_primary <- function() {
  return(c(blue = "#006DAE", black = "#000000", white = "#FFFFFF",
           gray80 = "#5A5A5A", gray50 = "#969696", gray10 = "#E6E6E6"))
}

#' @export
color_secondary <- function() {
  return(c(blue = "#027EB6", purple = "#746FB2", fuchsia = "#9651A0",
           ruby = "#C8008F", pink = "#ee64a4", red = "#EE0220",
           orange = "#D93F00", umber = "#795549", olive = "#6F7C4D",
           green = "#008A25"))
}


#' @export
color_info <- function() {
  # CLEAN
  cat("PRIMARY PALETTE

Monash Blue is our most identifiable colour. It conveys youthfulness, possibility and openness. It should always be considered first.

Black conveys prestige, timelessness and sophistication.

White is often shown as white space and conveys the brand personality of being open and youthful.

Greys can be any percentage of black and provides additional flexibility to our primary colour palette.

The primary colour palette is preferred for digital work.

  SECONDARY PALETTE

Inspired by the colours of our academic robes, we have a range of bright, colourful secondary colours.

Secondary colours are to be used:

* in charts and diagrams to highlight key findings
* as headings and subheadings
* sparingly to provide highlights or accents – ideally one or two secondary colours per double page spread
* to speak to a particular audience group. For instance, colours that would resonate best with our prospective undergraduate audience would be colourful, bright and youthful. Industry and research would suit a more corporate/mature choice to reflect focus and prestige.")
}


#' @family monash-color
#' @export
color_show_secondary <- function() {

}


#' Get Monash logo
#'
#' @description A quick way of getting the Monash logo.
#'
#' @param path the path to save the file to
#' @param stack TRUE for stacked logo, FALSE for one-line logo
#' @param color `blue`, `black` or `white`
#' @param type `png`, `jpg` or `ai`
#' @param hq TRUE for high-quality image (larger file size)
#' @param overwrite TRUE for overwriting (should it enquire?), FALSE for not. Not implemented yet.
#' @param filename A new file name for the logo. Not implemented yet.
#'
#' @name logo
#' @source https://www.monash.edu/brandbook/brand-elements/our-logo
NULL

#' @describeIn logo Download the logo to provided path
#' @export
logo_get <- function(path = ".",
                     stack = TRUE,
                     color = c("blue", "black", "white"),
                     type = c("png", "jpg", "ai"),
                     hq = FALSE,
                     overwrite = TRUE,
                     filename = NULL) {
  logo_path <- logo_path(stack = stack, color = color,
                         type = type, hq = hq)
  filename <- filename %||% path_file(logo_path)
  new_path <- path(path, filename)
  invisible(file_copy(logo_path, new_path = new_path, overwrite = overwrite))
  ui_done(glue("The {ui_value(filename)} is now in {ui_path(path)}"))
  new_path
}

#' @describeIn logo Get path to insert logo.
#' @importFrom vctrs vec_assert
#' @export
logo_path <- function(stack = TRUE,
                      color = c("blue", "black", "white"),
                      type = c("png", "jpg", "ai"),
                      hq = FALSE) {
  # validate arguments
  # all logo functions come here
  color <- arg_match(color)
  type <- arg_match(type)
  vec_assert(stack, logical(), size = 1)
  vec_assert(hq, logical(), size = 1)

  # get logo path
  color <- switch(color,
                  blue = ifelse(hq, "blue-cmyk", "blue-rgb"),
                  black = ifelse(hq, "black-cmyk", "black-rgb"),
                  white = ifelse(stack, "reversed-white", "reversed"))
  stack <- ifelse(stack, "stacked", "one-line")
  filename <- glue("monash-{stack}-{color}.{type}")
  logo <- logo_find(filename)
  logo
}



#' @describeIn logo A wrapper to insert logo.
#' @export
logo_insert <- function(path = NULL,
                        stack = TRUE,
                        color = c("blue", "black", "white"),
                        type = c("png", "jpg", "ai"),
                        hq = FALSE) {
  path <- path %||% rmarkdown::metadata$lib_dir %||% "."
  knitr::include_graphics(logo_get(path = path,
                                   stack = stack,
                                   color = color,
                                   type = type,
                                   hq = hq,
                                   overwrite = TRUE))
}


logo_find <- function(logo_filename) {
  path <- tryCatch(
    path_package(package = "monash", "logos", logo_filename),
    error = function(e) ""
  )
  if (identical(path, "")) {
    ui_stop(
      "Could not find logo {ui_value(logo_filename)}."
    )
  }
  path
}
